hadoop-ozone/dist/src/main/smoketest/httpfs/operations_tests.robot

```diff
@@ -0,0 +1,103 @@
+Resource            operations.robot
+
+*** Variables ***
+${URL}                         http://httpfs:14000/webhdfs/v1/
+${USERNAME}                    hdfs
+
+*** Test Cases ***
+Create volume
+    ${volume} =     Execute curl command    vol1    MKDIRS      -X PUT
+    Should contain  ${volume}   true
+
+Create first bucket
+    ${bucket} =     Execute curl command    vol1/buck1          MKDIRS      -X PUT
+    Should contain  ${bucket}   true
+
+Create second bucket
+    ${bucket} =     Execute curl command    vol1/buck2          MKDIRS      -X PUT
+    Should contain  ${bucket}   true
+
+Create local testfile
+    Create file       testfile
+
+Create testfile
+    ${file} =       Execute create file command     vol1/buck1/testfile     testfile
+    Should contain     ${file}     http://httpfs:14000/webhdfs/v1/vol1/buck1/testfile
+
+Read file
+    ${file} =       Execute curl command    vol1/buck1/testfile     OPEN    -L
+    Should contain     ${file}     Hello world!
+
+Delete bucket
+    ${bucket} =     Execute curl command    vol1/buck2          DELETE      -X DELETE
+    Should contain  ${bucket}   true
+
+Get status of bucket
+    ${status} =     Execute curl command    vol1/buck1          GETFILESTATUS      ${EMPTY}
+    Should contain  ${status}   FileStatus  DIRECTORY
+
+Get status of file
+    ${status} =     Execute curl command    vol1/buck1/testfile          GETFILESTATUS      ${EMPTY}
+    Should contain  ${status}   FileStatus  FILE    13
+
+List bucket
+    ${list} =       Execute curl command    vol1/buck1          LISTSTATUS      ${EMPTY}
+    Should contain  ${list}     FileStatus  testfile    FILE    13
+
+List file
+    ${list} =       Execute curl command    vol1/buck1/testfile          LISTSTATUS      ${EMPTY}
+    Should contain  ${list}     FileStatus  FILE    13
+
+List directory iteratively
+    ${list} =       Execute curl command    vol1          LISTSTATUS_BATCH&startAfter=buck1      ${EMPTY}
+    Should contain  ${list}     DirectoryListing    buck1
+    Should not contain          ${list}             buck2
+
+Get content summary of directory
+    ${summary} =    Execute curl command    vol1          GETCONTENTSUMMARY      ${EMPTY}
+    Should contain  ${summary}  ContentSummary      "directoryCount":2      "fileCount":1
+
+Get quota usage of directory
+    ${usage} =      Execute curl command    vol1          GETQUOTAUSAGE      ${EMPTY}
+    Should contain  ${usage}    QuotaUsage          "fileAndDirectoryCount":3
+
+Get home directory
+    ${home} =       Execute curl command    ${EMPTY}          GETHOMEDIRECTORY      ${EMPTY}
+    Should contain  ${home}     "Path":"\\/user\\/hdfs"
+
+Get trash root
+    ${trash} =      Execute curl command    vol1/buck1/testfile          GETTRASHROOT      ${EMPTY}
+    Should contain  ${trash}    "Path":"\\/vol1\\/buck1\\/.Trash\\/hdfs"
+
+Set permission of bucket
+    Execute curl command    vol1/buck1          SETPERMISSION&permission=7      -X PUT
```

Comment:
I believe the permission is a triplet of octal permissions here, so something like 755 or similar should be passed here as a parameter. Also we might want to test the current permissions returned are the default we expect before the test, and we should test if the permission is changed correctly after this test in a separate curl call. What do you think?

Reply:
Yes, it should be an octal, thank you for pointing it out. I modified the key words, so I was able to check the result json object of the list status operation. Even tho the results return code was 0 and the json returned with true, the change didn't happen. Thank you so much for the idea of testing the functionality other way. I commented these test cases until we fix the operation.